superset/db_engine_specs/postgres.py

```diff
@@ -92,6 +99,16 @@ class PostgresEngineSpec(PostgresBaseEngineSpec):
     try_remove_schema_from_table_name = False
 
     column_type_mappings = (
+        (
+            re.compile(r"^int.*", re.IGNORECASE),
+            INTEGER(),
+            utils.GenericDataType.NUMERIC,
+        ),
+        (
+            re.compile(r"^bool.*", re.IGNORECASE),
+            BOOLEAN(),
+            utils.GenericDataType.BOOLEAN,
+        ),
```

Comment:
I think these should be put in `BaseEngineSpec` (to replace `integer` and `boolean` respectively), as they're both fairly commonly used type names.

Reply:
This is specific to **Postgres Dialect** https://github.com/apache/superset/pull/13621/files#diff-8cec2642bdfebf02492bfb132c78133c229f409661a4e56c8ea96868a1afd63fR37
so keeping them separate like this might be better idea unless you think it's not needed?

Reply:
I know `INT` is at least used in [SQL Server](https://docs.microsoft.com/en-us/sql/t-sql/data-types/int-bigint-smallint-and-tinyint-transact-sql?view=sql-server-ver15) and a synonym on [Snowflake](https://docs.snowflake.com/en/sql-reference/intro-summary-data-types.html), and I don't know of any commonly used type names that would cause a collision with the shorter regexes, so I still feel like these should be in the base engine spec.

Reply:
Makes sense, will make a change